### PR TITLE
Feature/mediastore add getlocalmediaforpost

### PR DESCRIPTION
**Pull request by mzorz**
Adding `getMediaForPostWithState` and `getMediaForPost` methods to be able to get all media for a given post with a given state / unfiltered, correspondingly.

cc @aforcier

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java [hunk 1]
```diff
@@ -405,6 +405,17 @@ public MediaModel getMediaWithLocalId(int localMediaId) {
         return MediaSqlUtils.getSiteMediaWithIdsAsCursor(siteModel, mediaIds);
     }
 
+    public List<MediaModel> getMediaForPost(PostModel postModel) {
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java]: Would you mind grouping these with the other method using `matchPostMedia()` further down (`getPostMediaWithPath()`)?

Also we may as well name these consistently - I like yours best, so I'd say `s/getPostMediaWithPath/getMediaForPostWithState/`.

Comment (mzorz) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java]: Addressed in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/463/commits/2ca4dd1b49f1560b1fc9078606c03b62e48672ea

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java [hunk 2]
```diff
@@ -461,11 +461,22 @@ public String getThumbnailUrlForSiteMediaWithId(SiteModel siteModel, long mediaI
...
     }
 
-    public MediaModel getPostMediaWithPath(PostModel postModel, String filePath) {
+    public MediaModel getMediaForPostWithPath(PostModel postModel, String filePath) {
         List<MediaModel> media = MediaSqlUtils.matchPostMedia(postModel.getId(), MediaModelTable.FILE_PATH, filePath);
         return media.size() > 0 ? media.get(0) : null;
     }
 
+    public List<MediaModel> getMediaForPost(PostModel postModel) {
+        List<MediaModel> media = MediaSqlUtils.matchPostMedia(postModel.getId());
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java]: We don't need to create `media` here, we can just `return MediaSqlUtils.matchPostMedia(postModel.getId())` directly.

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java [hunk 3]
```diff
@@ -461,11 +461,22 @@ public String getThumbnailUrlForSiteMediaWithId(SiteModel siteModel, long mediaI
...
+        return media;
+    }
+
+    public List<MediaModel> getMediaForPostWithState(PostModel postModel, UploadState expectedState) {
+        List<MediaModel> media = MediaSqlUtils.matchPostMedia(postModel.getId(), MediaModelTable.UPLOAD_STATE,
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java]: Same as above comment, can `return` directly.

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java (hunk 2 above)
Comment (mzorz) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java]: addressed in 09ebd12

Comment (mzorz) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java]: addressed in 09ebd12

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java (hunk 3 above)
Comment (mzorz) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java]: addressed in 09ebd12

**Comment by mzorz** [comment]
Added WPAndroid counterpart PR here https://github.com/wordpress-mobile/WordPress-Android/pull/6034

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**